frontend/src/pages/register/form.tsx

```diff
@@ -1,76 +1,130 @@
-import React, { useState } from 'react'
+import React, { useState, useRef } from 'react'
+import { useForm, FormProvider, useFormContext } from 'react-hook-form'
 import { Form } from '../../components'
+import {
+  Error,
+  Input,
+  InputPassword,
+  PasswordContainer,
+  SeenIcon,
+} from '../../components/form/form-styles'
 import { Title } from '../../components/text/text-styles'
 
-const RegisterForm = () => {
-  const [email, setEmail] = useState<string>('')
-  const [password, setPassword] = useState<string>('')
-  const [confirmPassword, setConfirmPassword] = useState<string>('')
-  const [firstName, setFirstName] = useState<string>('')
-  const [lastName, setLastName] = useState<string>('')
+type Props = {
+  email: string
+  password: string
+  confirmPassword: string
+  firstName: string
+  lastName: string
+}
 
-  const emailChangeHandler = (event: React.FormEvent<HTMLInputElement>) => {
-    setEmail(event.currentTarget.value)
-    console.log(event.currentTarget.value)
-  }
+const RegisterForm = ({ loading, setFormData }: any) => {
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { errors },
+  } = useForm<Props>({})
 
-  const firstNameChangeHandler = (event: React.FormEvent<HTMLInputElement>) => {
-    setFirstName(event.currentTarget.value)
-    console.log(event.currentTarget.value)
-  }
-
-  const lastNameChangeHandler = (event: React.FormEvent<HTMLInputElement>) => {
-    setLastName(event.currentTarget.value)
-    console.log(event.currentTarget.value)
-  }
+  const password = useRef({})
+  // password.current(watch('password', ''))
+  const [passwordVisible, setPasswordVisible] = useState<boolean>(false)
+  const [confirmPasswordVisible, setConfirmPasswordVisible] =
+    useState<boolean>(false)
 
-  const passwordChangeHandler = (event: React.FormEvent<HTMLInputElement>) => {
-    setPassword(event.currentTarget.value)
-    console.log(event.currentTarget.value)
-  }
-
-  const confirmPasswordChangeHandler = (
-    event: React.FormEvent<HTMLInputElement>
-  ) => {
-    setConfirmPassword(event.currentTarget.value)
-    console.log(event.currentTarget.value)
+  const onSubmit = (data: any) => {
+    setFormData(data)
   }
 
   return (
     <>
       <Form>
         <Title fontSize={[4, 5, 6]}>Register</Title>
-        <Form.Error>Error</Form.Error>
-        <Form.Base>
-          <Form.Input
-            type="email"
+        <Form.Base onSubmit={handleSubmit(onSubmit)}>
+          <Input
+            {...register('email', {
+              required: 'This field is required',
```

Comment:
Extract this validation logic into a constant somewhere outside of the component, so that it is more readable